src/xclim/ensembles/_partitioning.py

```diff
@@ -302,6 +304,128 @@ def lafferty_sriver(
     return g, uncertainty
 
 
+def general_partition(
+    da: xr.DataArray,
+    sm: xr.DataArray | str = "loess",
+    var_first: list | None = None,
+    mean_first: list | None = None,
+    weights: list | None = None,
+) -> tuple[xr.DataArray, xr.DataArray]:
+    """
+    Return the mean and partitioned variance of an ensemble.
+
+    This is a general function that can be used to implemented methods from different papers.
+    The defaults are set to match the Lavoie et al. (2025, in preparation) method.
+
+    Parameters
+    ----------
+    da : xr.DataArray
+        Time series with dimensions 'time', mean_first and var_first.
+    sm : xr.DataArray or str
```

Comment:
I think you can provide a set of allowed values here:
```suggestion
    sm : xr.DataArray or {"poly", "loess"}
```